### PR TITLE
Improvement: Improvement 3dp left toolbar unified

### PR DESCRIPTION
**Pull request by MaShizhen**

### CODE REVIEW

File: src/app/models/ModelGroup.js [hunk 1]
```diff
@@ -2505,12 +2508,11 @@ class ModelGroup extends EventEmitter {
         return availModels;
     }
 
-    hasHideModel() {
```
Comment (wen-qing-255) [inline src/app/models/ModelGroup.js:2508]: why delete this function?

File: src/app/ui/widgets/PrintingVisualizer/Overlay/RotateOverlay.jsx [hunk 2]
```diff
@@ -167,7 +165,7 @@ const RotateOverlay = React.memo(({
                         priority="level-three"
                         width="100%"
                         onClick={autoRotate}
-                        disabled={hasHideModel}
+                        disabled={selectedModelArray.length > 0 && !isSelectedModelAllVisible}
```
Comment (wen-qing-255) [inline src/app/ui/widgets/PrintingVisualizer/Overlay/RotateOverlay.jsx:168]: Auto rotate disabled when all model are hidden.

Comment (wen-qing-255) [reply src/app/ui/widgets/PrintingVisualizer/Overlay/RotateOverlay.jsx:168]: We cannot delete `modelGroup.allIsHideModel`

File: src/app/ui/widgets/PrintingVisualizer/VisualizerLeftBar.jsx [hunk 3]
```diff
@@ -147,9 +148,7 @@ function VisualizerLeftBar({ setTransformMode, supportActions, updateBoundingBox
         },
     };
     // TODO: refactor these flags
-    const transformDisabled = showRotationAnalyzeModal || showEditSupportModal || !(selectedModelArray.length > 0 && selectedModelArray.every((model) => {
-        return model.visible === true;
-    }));
+    const transformDisabled = showRotationAnalyzeModal || showEditSupportModal || !hasModels;
```
Comment (wen-qing-255) [inline src/app/ui/widgets/PrintingVisualizer/VisualizerLeftBar.jsx:151]: why only selected primeTower will be transformDisabled status

File: src/app/models/ModelGroup.js [hunk 4]
```diff
@@ -2505,12 +2508,11 @@ class ModelGroup extends EventEmitter {
...
-        return _.some(this.selectedModelArray, { visible: false });
-    }
-
-    allIsHideModel() {
-        return _.every(this.selectedModelArray, { visible: false });
+    isSelectedModelAllVisible() {
+        if (this.selectedModelArray.length === 0) {
+            return false;
+        }
+        return this.selectedModelArray.every(model => model.visible && !(model instanceof PrimeTowerModel));
```
Comment (xiekun1992) [inline src/app/models/ModelGroup.js:2515]: consider to use `type`, eg: `this.modelGroup.selectedModelArray[0].type !== 'primeTower'`

File: src/app/models/ModelGroup.js [hunk 5]
```diff
@@ -599,6 +599,9 @@ class ModelGroup extends EventEmitter {
             if (selectModel) {
                 const objectIndex = this.selectedGroup.children.indexOf(selectModel.meshObject);
                 if (objectIndex === -1) {
+                    if (this.selectedModelArray.length === 1 && this.selectedModelArray[0] instanceof PrimeTowerModel) {
```
Comment (xiekun1992) [inline src/app/models/ModelGroup.js:602]: consider to use `type`, eg: `this.modelGroup.selectedModelArray[0].type !== 'primeTower'`

File: src/app/ui/components/SMCanvas/TransformControls.js [hunk 6]
```diff
@@ -955,6 +955,7 @@ class TransformControls extends Object3D {
     }
 
     onMouseMove(coord, isPrimeTower = false) {
+        console.log('isPrimeTower =', isPrimeTower);
```
Comment (xiekun1992) [inline src/app/ui/components/SMCanvas/TransformControls.js:958]: remove

**Review by xiekun1992 — CHANGES_REQUESTED** [review]

File: src/app/ui/widgets/PrintingVisualizer/Overlay/ExtruderOverlay.jsx [hunk 7]
```diff
@@ -160,15 +161,19 @@ const ExtruderOverlay = React.memo(({
...
-    const renderExtruderStatus = (status) => {
+    const renderExtruderStatus = (status, disabled) => {
+        if (!status && selectedModelArray.length === 0) {
+            status = '0';
+        }
         const leftExtruderColor = status === '1' ? colorR : colorL;
         const rightExtruderColor = status === '0' ? colorL : colorR;
         return (
-            <div className="sm-flex justify-space-between margin-left-16 width-160 border-default-black-5 border-radius-8 padding-vertical-4 padding-left-8">
+            <div className="sm-flex justify-space-between margin-left-16 width-160 border-default-black-5 border-radius-8 padding-vertical-4 padding-left-8" style={disabled ? { background: '#f5f5f5', cursor: 'not-allowed', color: 'rgba(0, 0, 0, 0.25)' } : {}}>
```
Comment (jane-rose) [inline src/app/ui/widgets/PrintingVisualizer/Overlay/ExtruderOverlay.jsx:171]: use default className

Comment (xiekun1992) [reply src/app/ui/widgets/PrintingVisualizer/Overlay/ExtruderOverlay.jsx:171]: +1

**Review by xiekun1992 — APPROVED** [review]

**Pull request merged**